installer/parse_install_yaml.py

```diff
@@ -11,13 +12,84 @@
 
 
 def show_error(error):
-    print("ERROR: {0}".format(error))
+    print(f"ERROR: {error}")
     return 1
 
 
+def type_git(install_item: Mapping, allowed_keys: Optional[List[str]] = None) -> str:
+    """
+    Function to check the parsed yaml for install type git and generate the command string.
+
+    The structure of a git install type is:
+    - type: git
+      url: <REPOSITORY_URL>
+      path: [LOCAL_CLONE_PATH]
+      version: [BRANCH_COMMIT_TAG]
+
+    :param install_item: Extracted yaml component corresponding to install type git
+    :param allowed_keys: Additional keys to allow apart from the keys defined in install type git
+    :return: Command string containing repository url and optional arguments target-dir and version
+    """
+
+    assert install_item["type"] == "git", "Invalid install type '{}' for type 'git'".format(install_item["type"])
+
+    keys = ["type", "url", "path", "version"]
+    if allowed_keys:
+        keys += allowed_keys
+
+    invalid_keys = [k for k in install_item.keys() if k not in set(keys)]
```

Comment:
Maybe better to have `keys` being a set from the start. Therefore `allowed_keys` should maybe also be a set. Or maybe allow it to be both.